midp/localmsg.js

```diff
@@ -1034,16 +1034,26 @@ MIDP.LocalMsgConnections["nokia.product-info"] = NokiaProductInfoLocalMsgConnect
 
 var localmsgServerWait = null;
 
+// XXX Consolidate the objects we store in this map with those in NativeMap.
+nativeConnectionMap = Object.create(null);
```

Comment:
> Is there a reason why you needed a different map or is it just a temporary thing?

It's just a temporary thing. I happened to create *nativeConnectionMap* for *connection* before I started to use *NativeMap* for the other two properties, which I bundled together into an *info* object. And it wasn't immediately clear how to integrate *connection* into the *info* object. So I left it alone.

Ideally, *server* and *protocolName* would both be stored in Java fields, since they're primitive values. So the only native will be *connection*. But I left that for a followup.